### PR TITLE
Email blacklist whitelist

### PR DESCRIPTION
**Pull request by carmenlau**
ref #906 

Completed:
- Support email blacklist / whitelist / block free email provider domains

TODO:
- Admin create user should skip blacklist / whitelist / block free email provider domains
- Enrich error's kind, so that `email domain is not allowed` and `username is not allowed` message can be translated in UI
- Update config in portal

### CODE REVIEW

File: pkg/lib/authn/identity/loginid/resources.go [hunk 1]
```diff
@@ -15,3 +16,26 @@ var ReservedNameTXT = resource.RegisterResource(resource.NewlineJoinedDescriptor
 		return blocklist.New(string(data))
 	},
 })
+
+var EmailBlacklistedDomainsTXT = resource.RegisterResource(resource.NewlineJoinedDescriptor{
+	Path: "email_blacklisted_domains.txt",
```
Comment (louischan-oursky) [inline pkg/lib/authn/identity/loginid/resources.go:21]: Let's name this `email_domain_blocklist.txt`

File: pkg/lib/authn/identity/loginid/resources.go [hunk 2]
```diff
@@ -15,3 +16,26 @@ var ReservedNameTXT = resource.RegisterResource(resource.NewlineJoinedDescriptor
...
+	Parse: func(data []byte) (interface{}, error) {
+		return exactmatchlist.New(string(data), true)
+	},
+})
+
+// FreeEmailProviderDomainsTXT is provided by
+// https://gist.github.com/tbrianjones/5992856/93213efb652749e226e69884d6c048e595c1280a
+var FreeEmailProviderDomainsTXT = resource.RegisterResource(resource.NewlineJoinedDescriptor{
+	Path: "free_email_provider_domains.txt",
```
Comment (louischan-oursky) [inline pkg/lib/authn/identity/loginid/resources.go:30]: Let's name this `free_email_provider_domain_list.txt`

File: pkg/lib/authn/identity/loginid/resources.go [hunk 3]
```diff
@@ -15,3 +16,26 @@ var ReservedNameTXT = resource.RegisterResource(resource.NewlineJoinedDescriptor
...
+	Parse: func(data []byte) (interface{}, error) {
+		return exactmatchlist.New(string(data), true)
+	},
+})
+
+var EmailWhiteListedDomainsTXT = resource.RegisterResource(resource.NewlineJoinedDescriptor{
+	Path: "email_whitlisted_domains.txt",
```
Comment (louischan-oursky) [inline pkg/lib/authn/identity/loginid/resources.go:37]: Let's name this `email_domain_allowlist.txt`

**Comment by carmenlau** [comment]
Replaced by PR #1083

**Pull request closed without merging**